src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -378,7 +378,7 @@ public function addFile(DatasetSubmission $datasetSubmission, Request $request,
             $datasetSubmission->setFileset($fileset);
         }
         $newFile = new File();
-        $newFile->setFileName(trim($request->get('name')));
+        $newFile->setFileName(trim(str_replace(':', '.', $request->get('name'))));
```

Comment:
I wouldn't do this here, or at all. The API should reject files with colons I feel. We can't change expectations.

Reply:
Most operating systems, don't allow file creation with ":". Atleast on macOS and windows, this is just an additional check William wanted